File: scenes/game/Level.fs
namespace BloodFeverFs.Scene

open BloodFeverFs.Actors
open Godot
open KSGodot
open KSGodotUtils

type Level() =
    inherit Base()

    member val ZombieTemplate = GD.Load<PackedScene>("res://actors/zombie/Zombie.tscn")

    member this.UI() = this.getNode ("UI")

    member this.Draco() = this.getNode<Actor> ("actors/Draco")

    member this.Actors() = this.getNode ("actors")

    member val stickInstance: Option<Stick> = None with get, set

    member this.Stick() =
        match this.stickInstance with
        | Some stick -> stick
        | None ->
            let s = (Stick(this.getNode ("UI/Stick")))
            this.stickInstance <- Some s
            s

    override this._Process(delta) =
        base._Process (delta)
        this.Draco().LinearVelocity <- this.Stick().Output() * 2f

    member this.OnSpawnTimer() =

        let mobSpawnLocation =
            this.getNode<PathFollow2D> ("SpawnPath/SpawnLocation")

        mobSpawnLocation.Offset <- float32 (this.NextRandom())

        let mobInstance = this.ZombieTemplate.Instance() :?> Actor
        this.Actors().AddChild(mobInstance)

        mobInstance.Position <- mobSpawnLocation.Position

File: scenes/game/Level.ts
import Actor from "../../actors/Actor";
import Base from "../Base";

export default class Level extends Base {
  ZombieTemplate: PackedScene<Actor> = load("res://actors/zombie/Zombie.tscn");

  get UI() {
    return this.get_node("UI");
  }

  get Draco() {
    return this.get_node("actors/Draco");
  }

  get Actors() {
    return this.get_node("actors");
  }

  get Stick() {
    return this.get_node("UI/Stick");
  }

  get spawnLocation() {
    return this.get_node("SpawnPath/SpawnLocation");
  }

  _process() {
    this.Draco.linear_velocity = this.Stick.output.mul(2);
  }

  onSpawnTimer() {
    this.spawnLocation.unit_offset = this.nextRandom();
    const mobInstance = this.ZombieTemplate.instance();
    this.Actors.add_child(mobInstance);
    mobInstance.position = this.spawnLocation.position;
  }
}
